src/apirequest.ts

```diff
@@ -228,18 +228,25 @@ async function createAPIRequestAsync<T>(parameters: APIRequestParams) {
 
   if (parameters.mediaUrl && media.body) {
     options.url = parameters.mediaUrl;
-    if (resource) {
+    if (resource || parameters.options.multipart === true) {
```

Comment:
To the point of your original question - do you know why we do this check for `resource` at all?  What would happen if we ... didn't?

Reply:
Some APIs support `multi-part`, and some apis don't (or at least, looking at discovery documents this is a field we can look at).

I think it would potentially be a bit risky to switch the behavior for all APIs to multipart. I think it would be good to see if this works for the user, and then consider either:

1. documenting this feature clearly, it actually seems like something where folks probably want to be able to control it rather than is inferring it, or:
2. setting this variable if the upstream API indicates it supports multi-part uploads.